website/pages/docs/from-1.5/special-variables.mdx

```diff
@@ -0,0 +1,63 @@
+---
+layout: docs
+page_title: Special Variables - HCL Configuration Language
+sidebar_title: Special Variables
+description: |-
+  Special variables provide connection information and basic instance state information.
+  This page covers all existing special variables.
+---
+
+# Build Variables
+
+`@include 'from-1.5/beta-hcl2-note.mdx'`
+
+Build variables will allow you to access, from provisioners and post-processors, connection information and basic instance state information.
+All special build variables are prefixed with `build`:
+
+```hcl
+  provisioner "shell" {
+      environment_vars = ["TESTVAR=${build.PackerRunUUID}"]
+      inline = ["echo $TESTVAR"]
+  }
+```
+
+The valid build variables to request are:
+
+- **ID**: Represents the vm being provisioned. For example, in Amazon it is the instance id; in digitalocean,
+  it is the droplet id; in Vmware, it is the vm name.
+
+- **Host**, **Port**, **User** and **Password**: The host, port, user, and password that Packer uses to access the machine.
+  Useful for using the shell local provisioner to run Ansible or Inspec against the provisioned instance.
+
+- **ConnType**: Type of communicator being used. For example, for SSH communicator this will be "ssh".
+
+- **PackerRunUUID**: Current build's unique id. Can be used to specify build artifacts.
+
+- **PackerHTTPIP**, **PackerHTTPPort**, and **PackerHTTPAddr**: HTTP IP, port, and address of the file server Packer creates to serve items in the "http" dir to the vm. The HTTP address is displayed in the format `IP:PORT`.
+
+- **SSHPublicKey** and **SSHPrivateKey**: The public and private key that Packer uses to connect to the instance.
+  These are unique to the SSH communicator and are unset when using other communicators.
+  **SSHPublicKey** and **SSHPrivateKey** can have escape sequences and special characters so their output should be single quoted to avoid surprises. For example:
+
+  ```hcl
+    provisioner "shell" {
+        inline = ["echo '${build.SSHPrivateKey}' > /tmp/packer-session.pem"]
+    }
+  ```
+
+For backwards compatability, `WinRMPassword` is also available through this
```

Comment:
```suggestion
For backwards compatibility, `WinRMPassword` is also available through this
```
I always write compatability too lol

Reply:
I just copied and pasted it from JSON so someone else also has the same problem! But if had written I would have done the same, because I just can't spell it right too 🤦